### PR TITLE
shared-module/audiomixer: expose MixerVoice.loop property

### PR DESCRIPTION
**Pull request by pdw-mb**
This PR exposes `MixerVoice.loop` as a mutable Python property.

The motivation here is to make it possible to stop a looping sample at the end of its current loop.  Without this, it is only possible to use `stop()` to stop the sample immediately.

My specific use case is a backing sound loop for a game, which then needs to transition into another sample. Without this feature,  I can't use the loop option, as I can't stop the sample at the end of a loop. With this feature, I can set `voice.loop = False`, then poll `voice.playing` to detect the end of the loop and start the next sample. It'd be nice to able to queue up the next sample to remove the need to poll, but that's a bigger change.

### CODE REVIEW

File: shared-bindings/audiomixer/MixerVoice.c [hunk 1]
```diff
@@ -107,6 +107,34 @@ MP_PROPERTY_GETSET(audiomixer_mixervoice_level_obj,
...
+static mp_obj_t audiomixer_mixervoice_obj_get_loop(mp_obj_t self_in) {
+    return mp_obj_new_bool(common_hal_audiomixer_mixervoice_get_loop(self_in));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(audiomixer_mixervoice_get_loop_obj, audiomixer_mixervoice_obj_get_loop);
+
+static mp_obj_t audiomixer_mixervoice_obj_set_loop(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_loop };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_loop,     MP_ARG_BOOL | MP_ARG_REQUIRED, {} },
+    };
```
Comment (jepler) [inline shared-bindings/audiomixer/MixerVoice.c:121]: for a property setter, there's no need to support kwargs. I notice that you were probably just copying the style of `audiomixer_mixervoice_obj_set_level` within this same file so what you wrote is understandable, but would be better to write it similar to this more usual setter:
```c
static mp_obj_t synthio_synthesizer_obj_set_envelope(mp_obj_t self_in, mp_obj_t envelope) {
    synthio_synthesizer_obj_t *self = MP_OBJ_TO_PTR(self_in);
    check_for_deinit(self);
    synthio_synth_envelope_set(&self->synth, envelope);
    return mp_const_none;
}
MP_DEFINE_CONST_FUN_OBJ_2(synthio_synthesizer_set_envelope_obj, synthio_synthesizer_obj_set_envelope);
```

**Review by jepler — CHANGES_REQUESTED** [review]
Thanks! This seems like a good idea. I have some comments, though.

File: shared-bindings/audiomixer/MixerVoice.c (hunk 1 above)
Comment (pdw-mb) [reply shared-bindings/audiomixer/MixerVoice.c]: Ah, I did wonder about that, and yes, very much a copy-and-paste job.  Fixed in 372759f728.

**Comment by jepler** [comment]
Thanks! One particular board build is failing, but this is not an error in your change. Rather, the board's flash space is full. We'll have to do something to shrink that build before we can pull in this PR, so thank you for being patient about that.

**Review by jepler — APPROVED** [review]
thank you!

**Pull request merged**